src/ctelnet.cpp

```diff
@@ -1908,3 +1910,34 @@ void cTelnet::raiseProtocolEvent(const QString& name, const QString& protocol)
     event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
     mpHost->raiseEvent(event);
 }
+
+// credit: https://github.com/qflow/websockets
+void cTelnet::setKeepAlive(int socketHandle)
+{
+#ifdef Q_OS_WIN
+    struct tcp_keepalive
+    {
+        u_long onoff;
+        u_long keepalivetime;
+        u_long keepaliveinterval;
+    } alive;
+    alive.onoff = TRUE;
+    alive.keepalivetime = 2 * 60 * 1000;
+    alive.keepaliveinterval = 3000;
+    DWORD dwBytesRet = 0;
+    int res = WSAIoctl(socketHandle, SIO_KEEPALIVE_VALS, &alive, sizeof(alive), NULL, 0, &dwBytesRet, NULL, NULL);
```

Comment:
Similarly to the `TRUE` above, is `NULL` okay here or should they be `nullptr`?

Reply:
I'm not sure - if you can understand https://msdn.microsoft.com/en-us/library/windows/desktop/ms741621(v=vs.85).aspx... otherwise I'm inclined to leave it as-is :laughing: